### PR TITLE
More miscellaneous Qt fixes

### PR DESCRIPTION
**Pull request by darkengine-io**
All the other stuff I have deserves its own PR, so this is probably the last 'misc fixes' PR for a while.

This PR contains the following:

* Fix a crash that would occur if a user Alt+F4'd with a game running
* Size the window up to fit the game if it's smaller than the IR (the user can still scale the window up or down once the game launches if they so choose)
* Fix the About Qt menu item not actually doing anything
* Bring in keyboard shortcuts from the wx UI.

What I'm working on next in rough order:

* Save states
* Movies
* Config window
* TAS (might need some help with this as I'm not too familiar with what a good TAS setup looks like)

### CODE REVIEW

**Comment by sigmabeta** [comment]
@buildbot rebuild

**Comment by JMC47** [comment]
@dolphin-emu-bot rebuild

**Comment by mathieui** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/DolphinQt/MainWindow.cpp [hunk 1]
```diff
@@ -92,6 +93,10 @@ DMainWindow::~DMainWindow()
 {
 }
 
+void DMainWindow::closeEvent(QCloseEvent *ce){
```
Comment (lioncash) [inline Source/Core/DolphinQt/MainWindow.cpp]: brace on next line

File: Source/Core/DolphinQt/MainWindow.cpp [hunk 2]
```diff
@@ -111,9 +116,15 @@ void DMainWindow::StartGame(const QString filename)
 		m_ui->centralWidget->addWidget(m_render_widget.get());
 		m_ui->centralWidget->setCurrentWidget(m_render_widget.get());
 
-		// TODO: When rendering to main, this won't resize the parent window...
-		m_render_widget->resize(SConfig::GetInstance().m_LocalCoreStartupParameter.iRenderWindowWidth,
-			SConfig::GetInstance().m_LocalCoreStartupParameter.iRenderWindowHeight);
+		if(SConfig::GetInstance().m_LocalCoreStartupParameter.bRenderWindowAutoSize)
```
Comment (lioncash) [inline Source/Core/DolphinQt/MainWindow.cpp]: space between the if and the (

File: Source/Core/DolphinQt/MainWindow.cpp [hunk 3]
```diff
@@ -111,9 +116,15 @@ void DMainWindow::StartGame(const QString filename)
...
+		{
+			// Resize main window to fit render
+			m_render_widget->setMinimumSize(SConfig::GetInstance().m_LocalCoreStartupParameter.iRenderWindowWidth,
+				SConfig::GetInstance().m_LocalCoreStartupParameter.iRenderWindowHeight);
+			qApp->processEvents(); // Force a redraw so the window has time to resize
+			m_render_widget->setMinimumSize(0,0); // Allow the widget to scale down
```
Comment (lioncash) [inline Source/Core/DolphinQt/MainWindow.cpp]: space after the comma.

File: Source/Core/DolphinQt/MainWindow.h [hunk 4]
```diff
@@ -58,11 +58,13 @@ private slots:
 	void OnOpenGitHub();
 	void OnOpenSystemInfo();
 	void OnOpenAbout();
+	void OnOpenAboutQt();
 
 	// Misc.
 	void UpdateIcons();
 
 private:
+	void closeEvent(QCloseEvent *ce);
```
Comment (lioncash) [inline Source/Core/DolphinQt/MainWindow.h]: Asterisk against type.

**Comment by mathieui** [comment]
@dolphin-emu-bot rebuild

**Comment by darkengine-io** [comment]
Is this an issue with FifoCI? Nothing here should affect rendering...

**Comment by BhaaLseN** [comment]
The EC2 instances were stuck for quite some time, so they seem to be a bit wonky now that they had some vacation.

**Pull request merged**